helm/dagger/templates/engine-statefulset.yaml

```diff
@@ -80,12 +90,28 @@ spec:
               name: {{ include "dagger.fullname" . }}-magicache-token
               {{- end }}
           {{- end }}
+          {{- if .Values.engine.port }}
+          ports:
+            - name: dagger
+              containerPort: {{ .Values.engine.port }}
+              protocol: TCP
+              {{- if .Values.engine.hostPort }}
+              hostPort: {{ .Values.engine.hostPort }}
+              {{- end }}
+          {{- end }}
```

Comment:
I really like how you layered the `hostPort` inside the port declaration.

Do you think that we should add a custom `--addr` arguments when this is defined @jholm117? I'm thinking:

```yaml
          args:
            - "--addr"
            - "tcp://0.0.0.0:{{ .Values.engine.port }}"
            - "--addr"
            - "unix:///var/run/buildkit/buildkitd.sock"
```

WDYT @jedevc ?

Reply:
I thought about that and decided to leave it out since I could already define it via `Values.args` and wanted to keep it simple. I think at the very least I can add a comment in the values file next to port that instructs the user to include the `--addr tcp://...` arg. Up to you!

Reply:
I think that we should add these values explicitly when and the port value is defined, otherwise users get caught out when the unix port is not specified explicitly and the health checks start failing. At least that was my experience many months ago when I forgot about the need to add `--addr unix:///var/run/buildkit/buildkitd.sock` explicitly.

Reply:
I had the same experience 😂 updated!

Reply:
Please look at https://github.com/dagger/dagger/pull/9866, the path will be change to `unix:///run/dagger/engine.sock` (backward compatibility is handled but might be fine to update that PR if it lands after 9866 is merged.

Reply:
Should it be `unix:///run/dagger/engine.sock` or `unix:///var/run/dagger/engine.sock`?

Reply:
<img width="926" alt="Screenshot 2025-03-17 at 19 57 03" src="https://github.com/user-attachments/assets/bec912ed-b815-4273-a7ea-dc7a50fc14fb" />

I think `/var/bin/dagger/engine.sock`

Reply:
Sigh, actually, no, this can't be removed because of a bug in how that engine socket is mounted.

Will open a separate PR to fix that.